source/base/function_lib.cc

```diff
@@ -1941,7 +1941,7 @@ namespace Functions
   FourierCosineFunction<dim>::value(const Point<dim>  &p,
                                     const unsigned int component) const
   {
-    (void)component;
+    [[mabe_unused]]component;
```

Comment:
This is the wrong place to add the attribute. You want to attach the attribute to the place where the variable is *declared*. So this should look as follows instead:
```
FourierCosineFunction<dim>::value(const Point<dim>  &p,
                                    [[maybe_unused]] const unsigned int component) const
  {
```
Can you change that here and elsewhere?